src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -585,4 +616,82 @@ private String mapToType(Reason reason) throws CTPException {
         throw new CTPException(Fault.SYSTEM_ERROR, "Unexpected refusal reason: %s", reason);
     }
   }
+
+  /**
+   * Make Case Service request to return cases by UPRN
+   *
+   * @param uprn of requested cases
+   * @param listCaseEvents boolean of whether require case events
+   * @return List of cases for UPRN
+   * @throws CTPException
+   */
+  private List<CaseDTO> callCaseSvcByUPRN(Long uprn, Boolean listCaseEvents) throws CTPException {
+
+    List<CaseContainerDTO> rmCases = new ArrayList<>();
+    try {
+      rmCases = caseServiceClient.getCaseByUprn(uprn, listCaseEvents);
+    } catch (ResponseStatusException ex) {
+      if (ex.getStatus() == HttpStatus.NOT_FOUND) {
+        log.with(uprn).info("Case by UPRN Not Found calling Case Service");
+        return Collections.emptyList();
+      } else {
+        log.with(uprn).with("status", ex.getStatus()).error("Error calling Case Service");
+        throw ex;
+      }
+    }
+
+    // Only return cases that are not of caseType = HI
+    List<CaseContainerDTO> casesToReturn =
+        (List<CaseContainerDTO>)
+            rmCases
+                .parallelStream()
+                .filter(c -> !(c.getCaseType().equals(CaseType.HI.name())))
+                .collect(Collectors.toList());
+
+    // Convert from Case service to Contact Centre DTOs
+    List<CaseDTO> caseServiceResponse = mapCaseContainerDTOList(casesToReturn);
+
+    // Clean up the events before returning them
+    caseServiceResponse.stream().forEach(c -> filterCaseEvents(c, listCaseEvents));
+    return caseServiceResponse;
+  }
+
+  /**
+   * Create new case, publish new address reported event and store new case in rerpository cache for
+   * new cases.
+   *
+   * @param uprn of address found
+   * @return Cachedcase details
+   * @throws CTPException
+   */
+  private CachedCase createNewCase(Long uprn) throws CTPException {
+
+    // Query AIMS for UPRN
+    Optional<AddressIndexAddressSplitDTO> addressQuery = addressSvc.uprnQuery(uprn);
+    if (addressQuery.isEmpty()) {
+      log.with("uprn", uprn).error("Failed to find address for UPRN");
+      throw new CTPException(Fault.RESOURCE_NOT_FOUND, "Failed to find address for UPRN %s", uprn);
+    }
+    AddressIndexAddressSplitDTO address = addressQuery.get();
+    if (address.getCountryCode() == SCOTTISH_COUNTRY_CODE) {
+      log.with("uprn", uprn)
+          .with("countryCode", address.getCountryCode())
+          .warn("Scottish address retrieved");
+      throw new CTPException(Fault.VALIDATION_FAILED, "Scottish address found for UPRN: " + uprn);
+    }
+
+    String newCaseId = (UUID.randomUUID().toString());
+    publishNewAddressReportedEvent(newCaseId, address);
```

Comment:
Minor point. More type safe signature to publishNewAddressReportedEvent would take a UUID as first param rather than string.

Reply:
As private, not public method within class thought OK. String variable is used in logging and construction of skeleton CachedCase stored in repository. Just saved a line of code and ensured the same value was used by having the string passed in the publishNewAddressReportedEvent method signature and used in CachedCase creation. I could not use a UUID in the CachedCase Id attribute as it does not have a no argument constructor and this caused a problem storing a UUID in Firestore. Will change if you feel strongly?

Reply:
Yes please . Always use a more specific type when one exists.

Reply:
Done.